### PR TITLE
test(stores): add comprehensive unit tests for Pinia stores

### LINKED ISSUE #56: Add unit tests for Pinia stores
## Summary
Add comprehensive unit tests for all Pinia stores created in #53.

## Stores requiring tests
- `buildStore` - complex serialization/deserialization logic, CharacterClass enum conversion
- `configStore` - mutual exclusivity rules for low/full life states
- `treeStore` - node lookups, selection state management
- `itemStore` - slot management, recent items handling
- `skillStore` - gem search functionality
- `uiStore` - notification lifecycle, preferences persistence

## Priority test areas
1. **buildStore serialization**: `toStoredBuild()` and `fromStoredBuild()` round-trip
2. **configStore mutual exclusivity**: toggle functions and setConfig behavior
3. **Character class conversion**: both string names and numeric values

## Acceptance criteria
- [ ] All store actions have unit tests
- [ ] Edge cases for enum conversions covered
- [ ] Mutual exclusivity rules verified
- [ ] Serialization/deserialization round-trips tested

Ref: PR #53

### PR DESCRIPTION
**Pull request by polaz**
## Summary
Add comprehensive unit tests for all 6 Pinia stores created in PR #53.

## Changes
- **buildStore.test.ts** (20 tests): Tests build state management, CharacterClass enum handling, node allocation/deallocation, skill group management, import/export
- **configStore.test.ts** (29 tests): Tests mutual exclusivity of low/full life states, charge management, condition toggles
- **treeStore.test.ts** (27 tests): Tests node lookups via Map, viewport management, search functionality, path highlighting, comparison mode
- **itemStore.test.ts** (28 tests): Tests slot management, recent items (FIFO with limit), clipboard operations, weapon swap
- **skillStore.test.ts** (28 tests): Tests gem search by name/tag, selection state, drag and drop state, group editor
- **uiStore.test.ts** (27 tests): Tests notification lifecycle with auto-dismiss, theme management, modal state, preferences

## Testing
- [x] All 216 tests pass
- [x] Lint passes with 0 errors
- [x] Build succeeds

Closes #56

### CODE REVIEW

File: test/unit/stores/itemStore.test.ts [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+    it('should have swap slots', () => {
+      expect(SWAP_SLOTS).toHaveLength(2);
+      expect(SWAP_SLOTS.find((s) => s.slot === ItemSlot.SLOT_WEAPON_1_SWAP)).toBeDefined();
+    });
+  });
+});
+
+// Need to import vi for mocking
+import { vi } from 'vitest';
```
[the remark this answers is not in the document]
Comment (polaz) [reply test/unit/stores/itemStore.test.ts]: Fixed in commit 6628a88. Moved `vi` import to line 7 with other vitest imports and removed the duplicate import at line 314.

File: test/unit/stores/skillStore.test.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+      store.closeGroupEditor();
+
+      expect(store.isGroupEditorOpen).toBe(false);
+    });
+  });
+
+  describe('gem selector', () => {
+    it('should open gem selector and clear search', () => {
+      const store = useSkillStore();
+      store['gemSearchQuery'] = 'old query';
```
[the remark this answers is not in the document]
Comment (polaz) [reply test/unit/stores/skillStore.test.ts]: Fixed in commit 6628a88. Now uses `store.setGemSearchQuery("old query")` instead of bracket notation as suggested.
